Content.Server/Nutrition/EntitySystems/FoodSystem.cs

```diff
@@ -115,6 +116,9 @@ public bool TryUseFood(EntityUid uid, EntityUid userUid, FoodComponent? componen
             if (!Resolve(uid, ref component))
                 return false;
 
+            if (EntityManager.TryGetComponent<MobStateComponent>(uid, out var mobState) && mobState.IsAlive())
+                return false;
+
```

Comment:
Would be much better to check the target is yourself to avoid any potential future issues (e.g. a chocolate bar eating itself)